xdsl/ir/core.py

```diff
@@ -761,6 +761,17 @@ def __iter__(self) -> Iterator[SSAValue]:
     def __len__(self) -> int:
         return len(self._op._operands)  # pyright: ignore[reportPrivateUsage]
 
+    def __eq__(self, other: object):
+        if not isinstance(other, OpOperands):
+            return False
+        return (
+            self._op._operands  # pyright: ignore[reportPrivateUsage]
+            == other._op._operands  # pyright: ignore[reportPrivateUsage]
+        )
+
+    def __hash__(self):
```

Comment:
Does this already work for a specific operand and not a list of Operands?

Reply:
Yep, Operands, Blocks, Regions, and Values already implement eq via pointer equality, and hash via pointer to the object

Reply:
Thanks!